google-cloud-clients/google-cloud-storage/src/main/java/com/google/cloud/storage/BucketInfo.java

```diff
@@ -95,6 +95,118 @@ public BucketInfo apply(com.google.api.services.storage.model.Bucket pb) {
+   */
+  public static class IamConfiguration implements Serializable {
+    private static final long serialVersionUID = -8671736104909424616L;
+
+    private Boolean isBucketPolicyOnlyEnabled;
+    private Long bucketPolicyOnlyLockedTime;
+
+    @Override
+    public boolean equals(Object o) {
+      if (this == o) return true;
+      if (o == null || getClass() != o.getClass()) {
+        return false;
+      }
+      IamConfiguration other = (IamConfiguration) o;
+      return Objects.equals(toPb(), other.toPb());
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hash(isBucketPolicyOnlyEnabled, bucketPolicyOnlyLockedTime);
+    }
+
+    private IamConfiguration(Builder builder) {
+      this.isBucketPolicyOnlyEnabled = builder.isBucketPolicyOnlyEnabled;
+      this.bucketPolicyOnlyLockedTime = builder.bucketPolicyOnlyLockedTime;
+    }
+
+    public static Builder newBuilder() {
+      return new Builder();
+    }
+
+    public Builder toBuilder() {
+      Builder builder = new Builder();
+      builder.isBucketPolicyOnlyEnabled = isBucketPolicyOnlyEnabled;
+      builder.bucketPolicyOnlyLockedTime = bucketPolicyOnlyLockedTime;
+      return builder;
+    }
+
+    public Boolean isBucketPolicyOnlyEnabled() {
+      return isBucketPolicyOnlyEnabled;
+    }
+
+    public Long getBucketPolicyOnlyLockedTime() {
+      return bucketPolicyOnlyLockedTime;
+    }
+
+    Bucket.IamConfiguration toPb() {
+      Bucket.IamConfiguration iamConfiguration = new Bucket.IamConfiguration();
+
+      Bucket.IamConfiguration.BucketPolicyOnly bucketPolicyOnly =
+          new Bucket.IamConfiguration.BucketPolicyOnly();
+      bucketPolicyOnly.setEnabled(isBucketPolicyOnlyEnabled);
+      bucketPolicyOnly.setLockedTime(
+          bucketPolicyOnlyLockedTime == null ? null : new DateTime(bucketPolicyOnlyLockedTime));
+
+      iamConfiguration.setBucketPolicyOnly(bucketPolicyOnly);
+
+      return iamConfiguration;
+    }
+
+    static IamConfiguration fromPb(Bucket.IamConfiguration iamConfiguration) {
+      Bucket.IamConfiguration.BucketPolicyOnly bucketPolicyOnly =
+          iamConfiguration.getBucketPolicyOnly();
+      DateTime lockedTime = bucketPolicyOnly.getLockedTime();
+
+      return newBuilder()
+          .setIsBucketPolicyOnlyEnabled(bucketPolicyOnly.getEnabled())
+          .setBucketPolicyOnlyLockedTime(lockedTime == null ? null : lockedTime.getValue())
+          .build();
+    }
+
+    /** Builder for {@code IamConfiguration} */
+    public static class Builder {
+      private Boolean isBucketPolicyOnlyEnabled;
+      private Long bucketPolicyOnlyLockedTime;
+
+      /**
+       * Sets whether BucketPolicyOnly is enabled for this bucket. When this is enabled, access to
+       * the bucket will be configured through IAM, and legacy ACL policies will not work. When this
+       * is first enabled, {@code lockedTime} will be set by the API automatically. This field can
```

Comment:
Could you use `bucketPolicyOnly.lockedTime` as it will help scope what this belongs within `IamConfiguration`.